File: app/src/main/java/com/yujin/fontchecker/model/ColorModel.java
package com.yujin.fontchecker.model;

import android.graphics.Color;
import android.widget.SeekBar;

import androidx.lifecycle.MutableLiveData;

public class ColorModel {
    public MutableLiveData<Integer> color = new MutableLiveData<>();
    public MutableLiveData<Integer> red = new MutableLiveData<>();
    public MutableLiveData<Integer> green = new MutableLiveData<>();
    public MutableLiveData<Integer> blue = new MutableLiveData<>();

    public ColorModel() {
        red.setValue(0);
        green.setValue(0);
        blue.setValue(0);
        color.setValue(Color.rgb(0,0,0));
    }

    public void onRedProgressChanged(SeekBar seekBar, int progressValue, boolean fromUser) {
        red.setValue(progressValue);
        color.setValue(Color.rgb(red.getValue(), green.getValue(), blue.getValue()));
    }

    public void onGreenProgressChanged(SeekBar seekBar, int progressValue, boolean fromUser) {
        green.setValue(progressValue);
        color.setValue(Color.rgb(red.getValue(), green.getValue(), blue.getValue()));
    }

    public void onBlueProgressChanged(SeekBar seekBar, int progressValue, boolean fromUser) {
        blue.setValue(progressValue);
        color.setValue(Color.rgb(red.getValue(), green.getValue(), blue.getValue()));
    }
}

File: app/src/main/java/com/yujin/fontchecker/model/ColorModel.kt
package com.yujin.fontchecker.model

import android.widget.SeekBar
import androidx.lifecycle.MutableLiveData
import com.yujin.fontchecker.util.RGB_COLOR_MIN
import com.yujin.fontchecker.util.getRGB

class ColorModel {
    val red = MutableLiveData(RGB_COLOR_MIN)
    val green = MutableLiveData(RGB_COLOR_MIN)
    val blue = MutableLiveData(RGB_COLOR_MIN)
    val color = MutableLiveData(RGB_COLOR_MIN)

    private val rgb: Int
        get() = getRGB(red.value!!, green.value!!, blue.value!!)

    fun onRedProgressChanged(seekBar: SeekBar, progressValue: Int, fromUser: Boolean) {
        red.value = progressValue
        color.value = rgb
    }

    fun onGreenProgressChanged(seekBar: SeekBar, progressValue: Int, fromUser: Boolean) {
        green.value = progressValue
        color.value = rgb
    }

    fun onBlueProgressChanged(seekBar: SeekBar, progressValue: Int, fromUser: Boolean) {
        blue.value = progressValue
        color.value = rgb
    }
}